addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -3452,6 +3452,52 @@ QUnit.module('relational_fields', {
         form.destroy();
     });
 
+    QUnit.test('add a new record on the first page of a X2many list with a second page', function (assert) {
+        assert.expect(2);
+
+        this.data.turtle.fields.partner_ids.type = 'one2many';
+        this.data.turtle.records[0].partner_ids = [1];
+        // we need a second page, so we set two records and only display one per page
+        this.data.partner.records[0].turtles = [1, 2];
+
+        this.data.partner.onchanges = {
+            turtles: function (obj) {
+                obj.display_name = obj.display_name + 1;
+                obj.turtles.splice(0, 0, [5]);
+                obj.turtles = obj.turtles.slice();
+        }};
+        this.data.turtle.onchanges = {
+            display_name: function (obj) {obj.turtle_foo = obj.display_name + 1;},
+        };
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<field name="display_name"/>' +
+                    '<field name="turtles">' +
+                        '<tree editable="bottom" limit="1">' +
+                            '<field name="display_name"/>' +
+                            '<field name="turtle_foo" required="True"/>' +
```

Comment:
Yes, it crashes without the fix.

Reply:
well, i don't understand why :-)

Reply:
Because the record hasn't been given any js id, so you basically have a null pointer exception (or whatever you would call that in jsspeak).
AFAIU that would be due to the test environment? 
So it crashes one or two lines above where it is supposed to crash.